internal/service/datalakepipeline/resource_data_lake_pipeline.go

```diff
@@ -475,208 +474,195 @@ func splitDataLakePipelineImportID(id string) (projectID, name string, err error
-	if atlasPipelineSink == nil {
+func flattenSink(sink *admin.IngestionSink) []map[string]any {
+	if sink == nil {
 		return nil
 	}
-
 	return []map[string]any{
 		{
-			"type":             atlasPipelineSink.Type,
-			"provider":         atlasPipelineSink.MetadataProvider,
-			"region":           atlasPipelineSink.MetadataRegion,
-			"partition_fields": flattenDataLakePipelinePartitionFields(atlasPipelineSink.PartitionFields),
+			"type":             sink.GetType(),
+			"provider":         sink.GetMetadataProvider(),
+			"region":           sink.GetMetadataRegion(),
+			"partition_fields": flattenPartitionFields(sink.GetPartitionFields()),
 		},
 	}
 }
 
-func flattenDataLakePipelineIngestionSchedules(atlasPipelineIngestionSchedules []*matlas.DataLakePipelineIngestionSchedule) []map[string]any {
-	if len(atlasPipelineIngestionSchedules) == 0 {
+func flattenIngestionSchedules(schedules []admin.DiskBackupApiPolicyItem) []map[string]any {
+	if len(schedules) == 0 {
 		return nil
 	}
-
-	out := make([]map[string]any, len(atlasPipelineIngestionSchedules))
-	for i, schedule := range atlasPipelineIngestionSchedules {
+	out := make([]map[string]any, len(schedules))
+	for i, schedule := range schedules {
 		out[i] = map[string]any{
-			"id":                 schedule.ID,
-			"frequency_type":     schedule.FrequencyType,
-			"frequency_interval": schedule.FrequencyInterval,
-			"retention_unit":     schedule.RetentionUnit,
-			"retention_value":    schedule.RetentionValue,
+			"id":                 schedule.GetId(),
+			"frequency_type":     schedule.GetFrequencyType(),
+			"frequency_interval": schedule.GetFrequencyInterval(),
+			"retention_unit":     schedule.GetRetentionUnit(),
+			"retention_value":    schedule.GetRetentionValue(),
 		}
 	}
-
 	return out
 }
 
-func flattenDataLakePipelineSnapshots(snapshots []*matlas.DataLakePipelineSnapshot) []map[string]any {
+func flattenSnapshots(snapshots []admin.DiskBackupSnapshot) []map[string]any {
 	if len(snapshots) == 0 {
 		return nil
 	}
-
 	out := make([]map[string]any, len(snapshots))
-	for i, snapshot := range snapshots {
+	for i := range snapshots {
+		snapshot := &snapshots[i]
 		out[i] = map[string]any{
-			"id":               snapshot.ID,
-			"provider":         snapshot.CloudProvider,
-			"created_at":       snapshot.CreatedAt,
-			"expires_at":       snapshot.ExpiresAt,
-			"frequency_yype":   snapshot.FrequencyType,
-			"master_key":       snapshot.MasterKeyUUID,
-			"mongod_version":   snapshot.MongodVersion,
-			"replica_set_name": snapshot.ReplicaSetName,
-			"type":             snapshot.Type,
-			"snapshot_type":    snapshot.SnapshotType,
-			"status":           snapshot.Status,
-			"size":             snapshot.StorageSizeBytes,
-			"policies":         snapshot.PolicyItems,
+			"id":               snapshot.GetId(),
+			"provider":         snapshot.GetCloudProvider(),
+			"created_at":       conversion.TimePtrToStringPtr(snapshot.CreatedAt),
```

Comment:
same as other comment, when converting, shouldn't we return a string instead of a string pointer?